### ISSUE TITLE
 PR "terminal-window: Factor CSDs into geometry hints #452" seems to break some things for me

### ISSUE THREAD

**Issue by L-U-T-i** [mate-desktop/mate-terminal#453]
#### Expected behaviour
1. When mate-terminal is updated and session restarted, terminl windows keep the sizes from the previous (saved) session
2. When resizing mate-terminal window, the pop-up with a current dimension is displayed

#### Actual behaviour
1. All restored terminal windows are very small (just after the first restart after update, this doesn't seem to be an issue anymore after that, so not that much annoying...)
2. There is no size displaying popup (when resizing the window) at all (quite annoying for me)

#### Steps to reproduce the behaviour
Install mate-terminal-1.28.0 and restart the session

#### MATE general version
1.28

#### Package version
1.28.0

#### Linux Distribution
Rocky Linux 9.3, custom build Mate Desktop packages

#### Link to bugreport of your Distribution (requirement)
no (packages not provided by the distro)

---

After I've applied the reverted PR #452 patch and rebuild, the issue has gone. So, it is definitely due to this PR.

**Comment by lukefromdc** [comment]
I assume this is in x11? What window manager?

**Comment by L-U-T-i** [comment]
Yes, it is X11. Window manager is marco (with a built-in Xpresent compositor), GTK theme is Blue Menta (with MATE icons and pointer).

Everything practically as by default, I've just changed icons and pointers to MATE ones (but I guess it is irrelevant).

**Comment by lukefromdc** [comment]
I just tested with BlueMenta using marco on x11 with session saving (which I normally do not use) and the terminals came back up the same size and position as before. Didn't check the size displaying popup but I normally use compiz not marco in x11.  I do use it there but never noticed in marco in the short times I am using marco for test purposes. Your session and mine are far enough apart for things to pass unnoticed on my end. I had to close my mate-wayland session and log back into the x11 session three different times looking for the terminal window not coming up it's save size. I had not RESIZED the terminals if that makes any difference.

We need someone else on the team to try and duplicate this, I was unable to

**Comment by L-U-T-i** [comment]
I'm not sure we understand each other. I am not complaining about terminal saving size - this has been an issue only the first time after installing a bew 1.28.0 release (with the commit mentioned), but works perfectly fine after.

What I like is a small grey pop-up (that appear somewhere around the middle of the terminal window in x11/marco only, but not in compiz!) with a terminal window size (in columns / rows, not pixels) while you are resizing a terminal window only. It is very useful for me if I'd like to have a terminal with a particular number of columns for instance.

Here for example:
![Screenshot](https://github.com/mate-desktop/mate-terminal/assets/50832205/884a5afb-2ab8-47d2-a146-6830d1a3a2ea)

**Comment by lukefromdc** [comment]
Wonder if something in the last commit is causing those values to not be integers? Out of time tonight though

**Comment by computersforpeace** [comment]
I could totally believe I've caused regressions. I noted in the original issue that I'm not really an expert here. You certainly won't offend me if you need to revert/drop my patch, temporarily or long-term.

I also only have intermittent time to spend on this sort of thing, so while I might try to see if I can help resolve the issues, I can't make any guarantees.

Sorry for any trouble here.

**Comment by davidfetter** [comment]
> #### Expected behaviour
> 
>     1. When mate-terminal is updated and session restarted, terminl windows keep the sizes from the previous (saved) session
> 
>     2. When resizing mate-terminal window, the pop-up with a current dimension is displayed
> 
> 
> #### Actual behaviour
> 
>     1. All restored terminal windows are very small (just after the first restart after update, this doesn't seem to be an issue anymore after that, so not that much annoying...)
> 
>     2. There is no size displaying popup (when resizing the window) at all (quite annoying for me)
> 
> 
> #### Steps to reproduce the behaviour
> 
> Install mate-terminal-1.28.0 and restart the session
> #### MATE general version
> 
> 1.28
> #### Package version
> 
> 1.28.0
> #### Linux Distribution
> 
> Rocky Linux 9.3, custom build Mate Desktop packages
> #### Link to bugreport of your Distribution (requirement)
> 
> no (packages not provided by the distro)
> 
> After I've applied the reverted PR #452 patch and rebuild, the issue has gone. So, it is definitely due to this PR.

I'm getting this stuff, too.

I get that testing graphical stuff is not easy, but whatever was happening that let this pass needs a pretty large improvement.

+1 for reverting the patch that caused the behavior.

**Comment by lukefromdc** [comment]
This suggests we need to limit the code from
https://github.com/mate-desktop/mate-terminal/pull/452
to Wayland sessions only, as that's the only place it is needed and we don't
yet have session saving support under Wayland to worry about so it doesn't
break anything there

**Comment by davidfetter** [comment]
> This suggests we need to limit the code from #452 to Wayland sessions only, as that's the only place it is needed and we don't yet have session saving support under Wayland to worry about so it doesn't break anything there

Reducing the surface of the patch seems like a great idea.

I'm surely not the first person who's brought up the testing thing, but that seems like it's broader in scope than this patch. Is there somewhere I can look to find previous discussions about testing things GUI, acceptance of patches influenced by same, etc. for the software and/or the project?

**Comment by davidfetter** [comment]
Any updates on actually fixing this?
